test-data/phony-guests/make-fedora-img.py

```diff
@@ -0,0 +1,369 @@
+            "/var/lib/rpm", "/usr/lib/rpm",
+            "/var/log/journal", "/boot/grub",
+        ]:
+            self.g.mkdir_p(d)
+
+        self.g.write("/etc/redhat-release", "Fedora release 14 (Phony)\n")
+        self.g.write("/etc/fedora-release", "Fedora release 14 (Phony)\n")
+        self.g.write("/etc/motd", "Welcome to Fedora 14 (Phony)\n")
+        self.g.write("/etc/sysconfig/network", "hostname=fedora.invalid\n")
+        self.g.upload("fedora.fstab", "/etc/fstab")
+        if os.path.exists("fedora.mdadm"):
+            self.g.upload("fedora.mdadm", "/etc/mdadm.conf")
+
+        self.g.upload(str(self.srcdir / "fedora.db"), "/var/lib/rpm/rpmdb.sqlite")
+        self.g.touch("/usr/lib/rpm/rpmrc")
+        self.g.write("/usr/lib/rpm/macros", "%_dbpath /var/lib/rpm\n%_db_backend sqlite\n")
+        self.g.upload(str(self.srcdir / "../binaries/bin-x86_64-dynamic"), "/bin/ls")
+        self.g.chmod(0o755, "/bin/ls")
+
+        self.g.tar_in(str(self.srcdir / "fedora-journal.tar.xz"), "/var/log/journal", compress="xz")
+
+        for path, content in [
+            ("/etc/test1", "ABCDEFG"),
+            ("/etc/test2", ""),
+            ("/etc/test3", "A\nB\nC\nD\nE\nF\n"),
+            ("/bin/test1", "ABCDEFG"),
+            ("/bin/test2", "ZXCVBNM"),
+            ("/bin/test3", "1234567"),
+            ("/bin/test4", ""),
+        ]:
+            self.g.write(path, content)
+
+        self.g.chown(10, 11, "/etc/test3")
+        self.g.chmod(0o600, "/etc/test3")
+        self.g.ln_s("/bin/test1", "/bin/test5")
+        self.g.mkfifo(0o777, "/bin/test6")
+        self.g.mknod(0o777, 10, 10, "/bin/test7")
+        self.g.touch("/boot/grub/grub.conf")
+
+        success("Common content populated")
+
+    # Public API
+    def run(self) -> None:
+        info(f"Building layout: {self.layout.upper()}")
+
+        if self.layout == "partitions":
+            self._build_partitions()
+        elif self.layout == "partitions-md":
+            self._build_partitions_md()
+        elif self.layout == "btrfs":
+            self._build_btrfs()
+        elif self.layout == "lvm-on-luks":
+            self._build_lvm_on_luks()
+        elif self.layout == "luks-on-lvm":
+            self._build_luks_on_lvm()
+        else:
+            error(f"Unknown layout: {self.layout}")
+
+        self._populate_common()
+
+        for img in self.images:
+            final = self.output_dir / img[:-2]
+            os.rename(img, final)
+            success(f"Image ready → {final.name}")
+
+# CLI
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Create Fedora phony guest images for libguestfs tests",
+        epilog="Layouts: partitions, partitions-md, btrfs, lvm-on-luks, luks-on-lvm",
+    )
+    parser.add_argument("layout", nargs="?", default=os.environ.get("LAYOUT"),
+                        help="Image layout (or $LAYOUT)")
+    parser.add_argument("srcdir", nargs="?", type=Path, default=os.environ.get("SRCDIR"),
```

Comment:
As the environment variables are no longer used (a welcome improvement) best to delete these unused defaults.